Added logic to save monitor secret while start

shtripat opened the pull request:
This is required for the case of pure import of a cluster
created outside of tendrl. In such a case we should figure out
and persist the monitor secret to central store, so that
further additions of mons to the cluster can utilize the
same value.

tendrl-node-id: Tendrl/ceph-integration#236
Signed-off-by: Shubhendu <shtripat@redhat.com>

r0h4n commented on tendrl/ceph_integration/manager/__init__.py:
```diff
@@ -84,6 +86,30 @@ def main():
 
+    # Save the monitor secret
+    gevent.sleep(5)
+    out, err, rc = cmd_utils.Command(
```
Why dont you put all this in the sds_sync thread?

shtripat replied:
The monitor key is something only once we need to save after import is successful. We dont need to keep this syncing again and again as this doesnt change once cluster is created.

If you are particular about sds_sync, may be we can add the code there with a check. If monitor doesnt exists already, we would sync else not.

Comments?

Commit:
Added mon key syncing as part of sds sync

This is required if a cluster is created outside tendrl and then
imported. The monitor secret key should be figured out and maintained
as its required while expansion of the cluster with more monitor nodes.

Signed-off-by: Shubhendu <shtripat@redhat.com>

shtripat commented:
Verified

r0h4n commented on tendrl/ceph_integration/sds_sync/__init__.py:
```diff
@@ -78,6 +80,36 @@ def _run(self):
+            if NS._int.client.read(
+                "clusters/%s/_mon_key" % NS.tendrl_context.integration_id
+            ):
+                pass
```
if the key exists, you should "return" instead of "pass" right?

shtripat replied:
We need to continue with other syncing stuff. If `_mon_key` just dont try to write it again.

r0h4n replied:
ok then remove the "pass", if EtcdKeyNotFound is raised, you have handled it

shtripat replied:
Ok

r0h4n commented on tendrl/ceph_integration/sds_sync/__init__.py:
```diff
@@ -78,6 +80,36 @@ def _run(self):
+            if rc != 0:
+                Event(
+                    Message(
+                        priority="warning",
```
this should be debug, the users dont need to know this

shtripat replied:
Done

r0h4n commented on tendrl/ceph_integration/sds_sync/__init__.py:
```diff
@@ -78,6 +80,36 @@ def _run(self):
+                    )
+                )
+            else:
+                mon_sec = out.split('\n')[1].strip().split(' = ')[1].strip()
```
check if out is not None

shtripat replied:
Done

shtripat commented:
Verified

r0h4n committed:
Merge branch 'develop' into import-mon-key
